src/server/xtra1.c

```diff
@@ -2728,7 +3019,7 @@ static void calc_bonuses(int Ind)
 	p_ptr->icky_wield = FALSE;
 
 	/* Priest weapon penalty for non-blessed edged weapons */
-    if ((p_ptr->pclass == CLASS_PRIEST) && (!p_ptr->bless_blade) &&
+    if ((p_ptr->cp_ptr->flags & CF_BLESS_WEAPON) && (!p_ptr->bless_blade) &&
```

Comment:
Again, comment makes no sense anymore. Functions can be good for expressing intentions.